plugins/database/mongodb/mongodb.go

```diff
@@ -135,17 +135,17 @@ func (m *MongoDB) CreateUser(ctx context.Context, statements dbplugin.Statements
 		Roles:    mongoCS.Roles.toStandardRolesArray(),
 	}
 
-	err = session.DB(mongoCS.DB).Run(createUserCmd, nil)
+	result := client.Database(mongoCS.DB).RunCommand(ctx, createUserCmd, nil)
 	switch {
-	case err == nil:
-	case err == io.EOF, strings.Contains(err.Error(), "EOF"):
-		// Call getConnection to reset and retry query if we get an EOF error on first attempt.
-		session, err := m.getConnection(ctx)
+	case result.Err() == nil:
+	case result.Err() == io.EOF, strings.Contains(result.Err().Error(), "EOF"):
+		// Call getConnection to reset and retry query if we get an ErrNoDocuments error on first attempt.
```

Comment:
Do we have any idea why this is, and why retrying exactly one more time is better? If so it would be good to document because its a slightly odd construct.

Reply:
I don't have enough experience with mongodb to estimate if this is necessary or not so I just decided to keep this initially. Since @divjotarora also agrees with you I decided to remove it now.